Lib/ufo2ft/fastUFO.py

```diff
@@ -192,32 +247,71 @@ def __init__(
             self.groups = reader.readGroups()
             self.kerning = reader.readKerning()
             self.lib = reader.readLib()
-            glyphset = reader.getGlyphSet()
-            for name in glyphset.keys():
-                glyph = Glyph(name, self)
-                glyphset.readGlyph(
-                    glyphName=name, glyphObject=glyph, pointPen=glyph)
-                self.glyphset[name] = glyph
+            defaultLayerName = reader.getDefaultLayerName()
+            for layerName in reader.getLayerNames():
+                glyphset = reader.getGlyphSet(layerName)
+                if layerName not in self.layers:
+                    self.newLayer(layerName)
+                for name in glyphset.keys():
+                    glyph = self.layers[layerName].newGlyph(name)
+                    glyphset.readGlyph(
+                        glyphName=name, glyphObject=glyph, pointPen=glyph)
+            self._layers.defaultLayer = self._layers[defaultLayerName]
+            self.data.fileNames = reader.getDataDirectoryListing()
+        if self._layers.defaultLayer is None:
+            layer = self.newLayer(DEFAULT_LAYER_NAME)
+            self._layers.defaultLayer = layer
 
         self.kerningGroupConversionRenameMaps = None
 
     def __contains__(self, name):
-        return name in self.glyphset
+        return name in self._layers.defaultLayer
 
     def __getitem__(self, name):
-        return self.glyphset[name]
+        return self.layers.defaultLayer[name]
 
     def __iter__(self):
         for name in self.keys():
-            yield self.glyphset[name]
+            yield self.layers.defaultLayer[name]
+
+    def __delitem__(self, name):
+        del self.layers.defaultLayer[name]
 
     def keys(self):
-        return self.glyphset.keys()
+        return self.layers.defaultLayer.keys()
+
+    def __len__(self):
+        return len(self.keys())
 
     def newGlyph(self, name):
-        glyph = Glyph(name, self)
-        self.glyphset[name] = glyph
-        return glyph
+        return self.layers.defaultLayer.newGlyph(name)
+
+    def newLayer(self, name):
+        return self.layers.newLayer(name)
+
+    @property
+    def layers(self):
+        return self._layers
```

Comment:
I don't see much use of this `layers` property. Is it to protect the attribute as read-only? We are adults. Using a property means every time one access this attribute there's an extra method call beside the attribute lookup.